### PR TITLE
Adjust schema to support enforcement of exclusive hsm group membership

### PR DESCRIPTION
**Pull request by bklei**
## Summary and Scope

The exclusiveGroup HSM label will be used when creating HSM groups to ensure xnames are only assigned to one tenant, in the absence of using HSM partitions.  Also adding validating webhook to enforce immutability of the field.

## Issues and Related PRs

* Resolves [CASMINST-4871](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4871)

## Testing

```
label = "pepsi"
description = ""
exclusiveGroup = "tapms-exclusive-group-label"
tags = [ "vcluster-pepsi",]

[members]
ids = [ "x0c3s2b0n0", "x0c3s3b0n0",]
```

```
ncn-m001-48d9c509:/home/bklein/oidc # kubectl apply -n tenants -f pepsi/tenant.yaml
Error from server (EnforceExclusiveHsmGroups field is immutable): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"tapms.hpe.com/v1alpha1\",\"kind\":\"Tenant\",\"metadata\":{\"annotations\":{},\"name\":\"vcluster-pepsi\",\"namespace\":\"tenants\"},\"spec\":{\"childnamespaces\":[\"slurm\",\"user\"],\"tenantname\":\"vcluster-pepsi\",\"tenantresources\":[{\"enforceexclusivehsmgroups\":false,\"hsmgrouplabel\":\"pepsi\",\"type\":\"compute\",\"xnames\":[\"x0c3s2b0n0\",\"x0c3s3b0n0\"]}]}}\n"}},"spec":{"tenantresources":[{"enforceexclusivehsmgroups":false,"hsmgrouplabel":"pepsi","type":"compute","xnames":["x0c3s2b0n0","x0c3s3b0n0"]}]}}
to:
Resource: "tapms.hpe.com/v1alpha1, Resource=tenants", GroupVersionKind: "tapms.hpe.com/v1alpha1, Kind=Tenant"
Name: "vcluster-pepsi", Namespace: "tenants"
for: "pepsi/tenant.yaml": admission webhook "vtenant.kb.io" denied the request: EnforceExclusiveHsmGroups field is immutable
```

### Tested on:

  * Virtual Shasta

### Test description:

Tested creation, modification, and delete.

## Risks and Mitigations

Low

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: api/v1alpha1/tenant_webhook.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *  OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+package v1alpha1
+
+import (
+	"errors"
+
```
Comment (cdelatte-hpe) [inline api/v1alpha1/tenant_webhook.go:31]: nit extra line

File: api/v1alpha1/webhook_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
```
Comment (cdelatte-hpe) [inline api/v1alpha1/webhook_suite_test.go:37]: nit extra line.  same w/ line 40.

**Review by cdelatte-hpe — COMMENTED** [review]
just a little nit.

File: api/v1alpha1/tenant_webhook.go (hunk 1 above)
Comment (bklei) [reply api/v1alpha1/tenant_webhook.go:31]: Good catch, had to look it up -- vscode adds this for readability.  first standard library imports, blank line, then third party imports (and then sorts them alphabetically in each group)..

File: api/v1alpha1/webhook_suite_test.go (hunk 2 above)
Comment (bklei) [reply api/v1alpha1/webhook_suite_test.go:37]: the operator-sdk/kubebuilder generates this guy, but vscode seems to like this as above (and doesn't change it when saving)...

**Review by cdelatte-hpe — APPROVED** [review]

**Review by rgross727 — APPROVED** [review]

**Pull request merged**